Automattic/wp-calypso
# Search: add support for adding wpcom plan to the shopping cart

#### Changes proposed in this Pull Request

Changes:
* enable adding Search for WP.com and AT sites

#### Testing instructions

_FULL flow:_
- go to `jetpack/connect/jepack_search`,  `/wpcom_search`, `/jepack_search/monthly`,  `/wpcom_search/monthly`, 
- start typing a WP.com site (without the product) and choose it from the selection
- proceed to checkout and purchase the product
- verify that a Thank you screen is present after the purchase
- verify that track events fire: `calypso_cart_product_add` with correct product slug prop, verify network request with the correct product info

_Testing adding a correct product:_
- /checkout/:site/jetpack_search|wpcom_search|jetpack_search_monthly|wpcom_search_monthly irrespective of the product slug, show correct product type in the cart (wpcom_search for AT and wpcom, and jetpack_search for JP)
